vertical-pod-autoscaler/e2e/updater.go

```diff
@@ -113,6 +181,106 @@ func setupHamsterDeployment(f *framework.Framework, cpu, memory string, replicas
 	return d, podList
 }
 
+func setupHamsterReplicationController(f *framework.Framework, cpu, memory string, replicas *int32) (*apiv1.ReplicationController, *apiv1.PodList) {
+	hamsterContainer := setupHamsterContainer(cpu, memory)
+	rc := framework.RcByNameContainer("hamster-rc", 3, "k8s.gcr.io/ubuntu-slim:0.1",
+		hamsterLabels, hamsterContainer, nil)
+	if replicas != nil {
+		rc.Spec.Replicas = replicas
+	}
+	rc.Namespace = f.Namespace.Name
+	err := testutils.CreateRCWithRetries(f.ClientSet, f.Namespace.Name, rc)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	err = waitForRCPodsRunning(f, rc)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	pods, err := getHamsterPods(f)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return rc, pods
+}
+
+func waitForRCPodsRunning(f *framework.Framework, rc *apiv1.ReplicationController) error {
+	return wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		podList, err := getHamsterPods(f)
+		if err != nil {
+			framework.Logf("Error listing pods, retrying: %v", err)
+			return false, nil
+		}
+		podsRunning := int32(0)
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == apiv1.PodRunning {
+				podsRunning += 1
+			}
+		}
+		return podsRunning == *rc.Spec.Replicas, nil
+	})
+}
+
+func setupHamsterJob(f *framework.Framework, cpu, memory string, replicas *int32) (*batchv1.Job, *apiv1.PodList) {
+	parallelism := int32(3)
+	if replicas != nil {
+		parallelism = *replicas
+	}
+	job := framework.NewTestJob("notTerminate", "hamster-job", apiv1.RestartPolicyOnFailure,
+		parallelism, parallelism, nil, 10)
+	job.Spec.Template.Spec.Containers[0] = setupHamsterContainer(cpu, memory)
+	job.Spec.Template.Labels["app"] = "hamster"
+	err := testutils.CreateJobWithRetries(f.ClientSet, f.Namespace.Name, job)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	err = framework.WaitForAllJobPodsRunning(f.ClientSet, f.Namespace.Name, job.Name, parallelism)
+	podList, err := framework.GetJobPods(f.ClientSet, f.Namespace.Name, job.Name)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return job, podList
+}
+
+func setupHamsterRS(f *framework.Framework, cpu, memory string, replicas *int32) (*appsv1.ReplicaSet, *apiv1.PodList) {
+	rs := framework.NewReplicaSet("hamster-rs", f.Namespace.Name, 3,
+		hamsterLabels, "", "")
+	if replicas != nil {
```

Comment:
Nit: I think it would be cleaner to use a construct similar to other functions so:
```
desiredReplicas := 3
if replicas != nill { desiredReplicas = *replicas }
...NewReplicaSet(..., desiredReplicas)

```

Reply:
Done